### ISSUE TITLE
Support yields on locks

### ISSUE THREAD

**Issue by fdtaillefer** [miketrethewey/sm-json-data#487]
Currently, `yields` (which represents the activation of game flages) can only be put on a node.

However, to accurately represent how Zebes awakens (see #486) we need both doors in Pit Room to yield the flag, but only after the lock has been set and unlocked.

Since it's possible to interact with at least the right door before then, and since even for the left door the flag is made active by unlocking the door (even though by happenstance you can't interact with that one without unlocking it), it would be more accurate if we were also able to represent the activation of a game flag by unlocking a lock, rather than by interacting with a node.

* Add the property in the schema
* Add the property in region-readme.md

Note: The actual Pit Room overhaul is out of scope for this, #486 can take care of it once this issue is resolved.
